### PR TITLE
Add support for integration events

### PR DESCRIPTION
**Pull request by IDoEverything**
# Summary
Adds support for the integration events.

# Notes
Very long overdue.

### CODE REVIEW

File: DSharpPlus/Clients/DiscordShardedClient.cs [hunk 1]
```diff
@@ -487,6 +487,10 @@ private void InternalSetup()
             this._applicationCommandCreated = new AsyncEvent<DiscordClient, ApplicationCommandEventArgs>("APPLICATION_COMMAND_CREATED", DiscordClient.EventExecutionLimit, this.EventErrorHandler);
             this._applicationCommandUpdated = new AsyncEvent<DiscordClient, ApplicationCommandEventArgs>("APPLICATION_COMMAND_UPDATED", DiscordClient.EventExecutionLimit, this.EventErrorHandler);
             this._applicationCommandDeleted = new AsyncEvent<DiscordClient, ApplicationCommandEventArgs>("APPLICATION_COMMAND_DELETED", DiscordClient.EventExecutionLimit, this.EventErrorHandler);
+            this._guildStickersUpdated = new AsyncEvent<DiscordClient, GuildStickersUpdateEventArgs>("GUILD_STICKERS_UPDATED", DiscordClient.EventExecutionLimit, this.EventErrorHandler);
```
Comment (Lulalaby) [inline DSharpPlus/Clients/DiscordShardedClient.cs:490]: Dup of #1007

File: DSharpPlus/Clients/DiscordClient.cs [hunk 2]
```diff
@@ -206,6 +206,9 @@ internal void InternalSetup()
             this._applicationCommandCreated = new AsyncEvent<DiscordClient, ApplicationCommandEventArgs>("APPLICATION_COMMAND_CREATED", EventExecutionLimit, this.EventErrorHandler);
             this._applicationCommandUpdated = new AsyncEvent<DiscordClient, ApplicationCommandEventArgs>("APPLICATION_COMMAND_UPDATED", EventExecutionLimit, this.EventErrorHandler);
             this._applicationCommandDeleted = new AsyncEvent<DiscordClient, ApplicationCommandEventArgs>("APPLICATION_COMMAND_DELETED", EventExecutionLimit, this.EventErrorHandler);
+            this._integrationCreated = new AsyncEvent<DiscordClient, IntegrationCreateEventArgs>("INTEGRATION_CREATED", EventExecutionLimit, this.EventErrorHandler);
+            this._integrationUpdated = new AsyncEvent<DiscordClient, IntegrationUpdateEventArgs>("INTEGRATION_UPDATED", EventExecutionLimit, this.EventErrorHandler);
+            this._integrationDeleted = new AsyncEvent<DiscordClient, IntegrationDeleteEventArgs>("INTEGRATION_UPDATED", EventExecutionLimit, this.EventErrorHandler);
```
Comment (Kaoticz) [inline DSharpPlus/Clients/DiscordClient.cs:211]: Did you mean `"INTEGRATION_DELETED"` instead?

File: DSharpPlus/Clients/DiscordShardedClient.cs (hunk 1 above)
Comment (Naamloos) [reply DSharpPlus/Clients/DiscordShardedClient.cs:490]: Seems like a minor thing duplicated. Won't be much of a problem merging I think.

**Review by Naamloos — APPROVED** [review]
Looks fine.

**Comment by Naamloos** [comment]
Waiting for someone elses two cents before merging, just in case I missed something.

**Review by Neuheit — COMMENTED** [review]
Remove the guild stickers updated event and we should be good.

File: DSharpPlus/Clients/DiscordShardedClient.cs [hunk 3]
```diff
@@ -488,6 +488,10 @@ private void InternalSetup()
...
```
Comment (Neuheit) [inline DSharpPlus/Clients/DiscordShardedClient.cs:491]: This event is initialized twice. It should be removed here.

**Review by Neuheit — CHANGES_REQUESTED** [review]

**Review by Neuheit — APPROVED** [review]

**Pull request merged**